docs/specifications/tests/Nameserver-TP/nameserver14.md

```diff
@@ -0,0 +1,87 @@
+# NAMESERVER14: Test for unknown version with unknown OPTION-CODE
+
+## Test case identifier
+
+**NAMESERVER14** 
+
+### Objective
+
+This test case actually combines the test options in test cases
+[NAMESERVER10](../nameserver10.md) and [NAMESERVER11](../nameserver11.md). 
+
+### Inputs
+
+"Child Zone" - The domain name to be tested.
+
+### Ordered description of steps to be taken to execute the test case
+
+1. Create an SOA query for the *Child Zone* with an OPT record with 
+   EDNS version set to "1" and  with EDNS OPTION-CODE set to an
+   unassigned value (see [IANA-DNSSYSTEM-PARAMETERS]) and
+   with no other EDNS options or flags. 
+
+2. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+3. For each name server in *Name Server IP* do:
+
+	1. Send the SOA query to the name server and collect the response.
+	2. If there is no DNS response, output *[NO_RESPONSE]* and go to
+      	next server.
+	3. Else, if the DNS response has the RCODE "FORMERR" then output
+      	*[NO_EDNS_SUPPORT]*.
+	4. Else, if the RCODE has "NOERROR", and has an "OPTION-CODE" and
+	"Version Number" present in the response,  then output
+	[UNKNOWN_OPTION_CODE_VERSION]. 
+	5. Else, if the DNS response has the RCODE "NOERROR" with an OPT record
+	with version greater than zero, then output [UNSUPPORTED_EDNS_VER]
+	6. Else, if the DNS response has the RCODE "NOERROR" and "OPTION-CODE"
+	present in the response then output [UNKNOWN_OPTION_CODE].
```

Comment:
We will never come to "6" since "4" will catch all cases that "6" would catch.

Reply:
4 becomes true only if both option-code and version number are present. 
5 and 6 tests them individually.

Reply:
@sandoche2k, the test `"Version Number" present in the response` is the same thing as saying that OPT record is present. If there is an OPTION-CODE there is an OPT record. The following two condition statements are indentical:
1. The response has an "OPTION-CODE" and "Version Number" is present in the response.
2. The response has an "OPTION-CODE".

Reply:
Maybe you are right. But, i understand it differently from reading the RFC 6891: 

An OPT RR has a fixed part and a variable set of options expressed as   {attribute, value} pairs. 

The version number is part of the OPT record TTL field (which is the fixed part fo the OPT RR) as explained in 6.1.3 section of the RFC 6891

The OPTION-CODE is part of the variable part as explained in 6.1.2 section of the RFC 6891

Reply:
I think it is quite clear from the RFC. The EDNS version sits in the TTL field, which is always there, or else we have a broken DNS  packet. The EDNS version is always defined if the OPT record is there. What we can test is the version value.

From 6.1.3 of https://tools.ietf.org/html/rfc6891
```
The extended RCODE and flags, which OPT stores in the RR Time to Live
   (TTL) field, are structured as follows:

                  +0 (MSB)                            +1 (LSB)
       +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
    0: |         EXTENDED-RCODE        |            VERSION            |
       +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
    2: | DO|                           Z                               |
       +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+

(...)

   VERSION
      Indicates the implementation level of the setter.  Full
      conformance with this specification is indicated by version '0'.
(...)
```

Reply:
You are both right but I am not sure you both answer the same question...

Perhaps that with a different approach it could be less confusing:

1/ Please do not use `"Version Number" present in the response`, because it is always present if you have an OPT record (you should say `Version number is greater than 0` as used elesewhere in the text).
2/ Use a matrix for step 4,5,6